src/engine/QueryPlanner.cpp

```diff
@@ -2624,6 +2635,10 @@ void QueryPlanner::GraphPatternPlanner::visitGroupOptionalOrMinus(
   // candidates.
   if (candidates[0].type == SubtreePlan::BASIC) {
     candidatePlans_.push_back(std::move(candidates));
+    for (auto& plan : candidatePlans_.back()) {
+      plan._idsOfIncludedFilters = 0;
+      plan.idsOfIncludedTextLimits_ = 0;
+    }
```

Comment:
Add a comment here "we have finished a nested GroupGraphPattern, reset the filter IDs, s.t. they don't leak into other groups"